### PR TITLE
BAC-2720: Add AWSRole deprecate AWSECSRole 

### PR DESCRIPTION
**Pull request by burakcive**

### CODE REVIEW

File: internal/provider/trust_provider_resource.go [hunk 1]
```diff
@@ -639,6 +676,8 @@ func convertTrustProviderDTOToModel(ctx context.Context, dto aembit.TrustProvide
 	switch dto.Provider {
 	case "AWSECSRole":
 		model.AwsEcsRole = convertAwsEcsRoleDTOToModel(dto)
+	case "AWSRole":
+		model.AwsRole = convertAwsEcsRoleDTOToModel(dto)
```
Comment (m43kwon) [inline internal/provider/trust_provider_resource.go:680]: Should we create a separate routine convertAwsRoleDTOToModel? Or rename convertAwsEcsRoleDTOToModel to convertAwsRoleDTOToModel?

**Review by m43kwon — APPROVED** [review]

File: internal/provider/trust_provider_resource.go [hunk 2]
```diff
@@ -110,7 +110,30 @@ func (r *trustProviderResource) Schema(_ context.Context, _ resource.SchemaReque
 				},
 			},
 			"aws_ecs_role": schema.SingleNestedAttribute{
-				Description: "AWS ECS Role type Trust Provider configuration.",
+				Description:        "AWS ECS Role type Trust Provider configuration.",
+				Optional:           true,
+				DeprecationMessage: "Deprecated. Use aws_role instead",
```
Comment (quantifex) [inline internal/provider/trust_provider_resource.go:115]:
```suggestion
				DeprecationMessage: "Please migrate to `aws_role` which replaces this Trust Provider type.",
```

File: internal/provider/trust_provider_resource.go [hunk 3]
```diff
@@ -110,7 +110,30 @@ func (r *trustProviderResource) Schema(_ context.Context, _ resource.SchemaReque
...
+						Description: "The UsernID of the AWS IAM Account which is running the ECS Task (not commonly used).",
+						Optional:    true,
+					},
+				},
+			},
+			"aws_role": schema.SingleNestedAttribute{
+				Description: "AWS Role type Trust Provider configuration.",
 				Optional:    true,
 				Attributes: map[string]schema.Attribute{
 					"account_id": schema.StringAttribute{
```
Comment (quantifex) [inline internal/provider/trust_provider_resource.go:139]: You have several references to ECS in the descriptions here which need to be cleaned up.

File: internal/provider/trust_providers_data_source.go [hunk 4]
```diff
@@ -93,7 +93,30 @@ func (d *trustProvidersDataSource) Schema(_ context.Context, _ datasource.Schema
 							},
 						},
 						"aws_ecs_role": schema.SingleNestedAttribute{
-							Description: "AWS ECS Role type Trust Provider configuration.",
+							Description:        "AWS ECS Role type Trust Provider configuration.",
+							Computed:           true,
+							DeprecationMessage: "Deprecated. Use aws_role instead",
```
Comment (quantifex) [inline internal/provider/trust_providers_data_source.go:98]: Please use the deprecation message I suggested above.

File: internal/provider/trust_provider_resource.go (hunk 3 above)
Comment (burakcive) [reply internal/provider/trust_provider_resource.go:139]: Where can I get the correct descriptions for aws_role type attributes?

**Review by quantifex — APPROVED** [review]

**Pull request merged**